programs/solana/programs/ics07-tendermint/src/lib.rs

```diff
@@ -0,0 +1,140 @@
+#![allow(unexpected_cfgs)]
+#![allow(deprecated)]
+#![warn(clippy::all)]
+#![allow(clippy::result_large_err)]
+
+use anchor_lang::prelude::*;
+
+pub mod error;
+pub mod helpers;
+pub mod instructions;
+pub mod state;
+pub mod types;
+
+use crate::state::{ClientData, ConsensusStateStore};
+
+declare_id!("8wQAC7oWLTxExhR49jYAzXZB39mu7WVVvkWJGgAMMjpV");
+
+pub use types::{ClientState, ConsensusState, UpdateClientMsg, MembershipMsg, MisbehaviourMsg};
+
+#[derive(Accounts)]
+#[instruction(chain_id: String)]
+pub struct Initialize<'info> {
+    #[account(
+        init,
+        payer = payer,
+        space = 8 + 1000,
+        seeds = [b"client", chain_id.as_bytes()],
+        bump
+    )]
+    pub client_data: Account<'info, ClientData>,
+    #[account(
+        init,
+        payer = payer,
+        space = 8 + 8 + 8 + 32 + 32, // discriminator + height + timestamp + root + next_validators_hash
```

Comment:
You don't need to calculate these spaces by hand. You can derive `InitSpace` and then:
```suggestion
        space = 8 + ClientData::INIT_SPACE
```
ditto on all others